### PR TITLE
#160451519 assign roles to all users

### PR DESCRIPTION
**Pull request by onwuvic**
#### What does this PR do?
This PR add roles to all users

#### Description of Task to be completed?
- Add role field to user model and migration
- Create seeder file to seed initial admin

#### How should this be manually tested?
- Create a user and the default role will be User

#### Any background context you want to provide?
N/A

#### What are the relevant pivotal tracker stories?
Story type: feature
Story title: Role-based access control functionality
Story ID: #160451519

#### Screenshots (if appropriate)
N/A

#### Questions:```
N/A

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -12,6 +12,7 @@
     "migrate-undo": "sequelize db:migrate:undo:all",
     "migrate": "sequelize db:migrate",
     "migrate-reset": "sequelize db:migrate:undo:all && sequelize db:migrate",
+    "seed:admin": "sequelize db:seed --seed server/seeders/20181017084637-create-admin.js",
```
Comment (onwuvic) [inline package.json:15]: This is used to seed the first initial admin

File: server/seeders/20181017084637-create-admin.js [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+    email: process.env.ADMIN_EMAIL,
+    password: bcrypt.hashSync(process.env.ADMIN_PASS, 10),
+    isVerified: true,
+    role: process.env.ADMIN_ROLE,
+    createdAt: new Date(),
+    updatedAt: new Date()
+  }], {}),
+
+  down: queryInterface => queryInterface.bulkDelete('Users', null, {})
+};
```
Comment (onwuvic) [inline server/seeders/20181017084637-create-admin.js]: Will add newline here

**Review by s-akinrele — APPROVED** [review]

**Comment by s-akinrele** [comment]
Fix your failing test

File: .env_sample [hunk 3]
```diff
@@ -31,3 +31,8 @@ TWITTER_BASE_URL=http://127.0.0.1:3000/api/v1
 TWITTER_API_KEY=abcde
 TWITTER_API_SECRET=abcde
 SESSION_SECRET=12345
+
+ADMIN_EMAIL=email
+ADMIN_USERNAME=username
+ADMIN_ROLE=admin
+ADMIN_PASS=password
```
Comment (onwuvic) [inline .env_sample]: add newline

File: server/controllers/userController.js [hunk 4]
```diff
@@ -36,7 +36,7 @@ export default class UserController {
       })
         .then((user) => {
           const token = generateToken({
-            id: user.dataValues.id, username: user.username, email: user.email
+            id: user.dataValues.id, username: user.username, email: user.email, role: user.role
```
Comment (onwuvic) [inline server/controllers/userController.js:39]: I added user role to the token

File: server/controllers/userController.js [hunk 5]
```diff
@@ -160,15 +160,17 @@ export default class UserController {
...
+          id, email, username, isVerified, role
         } = user.dataValues;
 
         if (!isVerified) {
           return res.status(403).json({ success: false, message: 'Please verify your account' });
         }
 
         if (bcrypt.compareSync(req.body.password, user.password)) {
-          const token = generateToken({ id, email, username });
+          const token = generateToken({
```
Comment (onwuvic) [inline server/controllers/userController.js:171]: I added user role to the token

**Review by slimsolz — COMMENTED** [review]
LGTM

**Pull request merged**